authentik/policies/types.py

```diff
@@ -39,12 +40,13 @@ def __init__(self, user: User):
     def set_http_request(self, request: HttpRequest):  # pragma: no cover
         """Load data from HTTP request, including geoip when enabled"""
         self.http_request = request
-        if not GEOIP_READER.enabled:
-            return
         client_ip = get_client_ip(request)
         if not client_ip:
             return
-        self.context["geoip"] = GEOIP_READER.city(client_ip)
+        if GEOIP_ENRICHER.enabled:
+            self.context["geoip"] = GEOIP_ENRICHER.city(client_ip)
+        if ASN_ENRICHER.enabled:
+            self.context["asn"] = ASN_ENRICHER.asn(client_ip)
```

Comment:
Can enrichers have a `get_context` method instead? This pattern is used here and in the enricher itself for `enrich_event` as well